### PR TITLE
Changed _make_iterator as you suggested

### PR DESCRIPTION
**Pull request by guberger**

### CODE REVIEW

File: src/rectangle.jl [hunk 1]
```diff
@@ -20,3 +20,7 @@ end
 function Base.issubset(a::HyperRectangle, b::HyperRectangle)
     return all(a.lb .>= b.lb) && all(a.ub .<= b.ub)
 end
+
+function name(args)
```
Comment (blegat) [inline src/rectangle.jl:24]: What is this ?

Comment (guberger) [reply src/rectangle.jl]: A function... Voilà c'est fixé ;-)

File: src/macros.jl [hunk 2]
```diff
@@ -78,3 +78,54 @@ function set_controller_reach!(sym_model_contr, sym_model_sys, X_init, X_target)
 	end
 	println("\nset_controller_reach! terminated with success")
 end
+
+# "Set" assumes sym_model_contr is empty initially... May fail if not respected
+function set_controller_safe!(sym_model_contr, sym_model_sys, X_init, X_safe)
+	@assert X_init.grid_space == X_target.grid_space ==
```
Comment (blegat) [inline src/macros.jl:84]: Uses `===` otherwise, it will be slow once we implement `==` between grid spaces

File: src/macros.jl [hunk 3]
```diff
@@ -78,3 +78,54 @@ function set_controller_reach!(sym_model_contr, sym_model_sys, X_init, X_target)
...
+	union_subsets!(X_init2, X_init)
+	X_target2 = NewSubSet(X_grid)
+	# sizehint_subset!(X_target2, size_remain)
+	union_subsets!(X_target2, X_target)
+	setdiff_subsets!(X_remain, X_target)
+	setdiff_subsets!(X_init2, X_target)
+	xref_new_coll = get_gridspace_reftype(sym_model_sys.X_grid)[]
+	uref_enabled_coll = get_gridspace_reftype(sym_model_sys.U_grid)[]
+
+	while ~is_subset_empty(X_init2)
```
Comment (blegat) [inline src/macros.jl:109]: Use `!` instead of `~`. `~` is for bitwise, e.g.
```julia
julia> ~8
-9
```

File: src/macros.jl [hunk 4]
```diff
@@ -78,3 +78,53 @@ function set_controller_reach!(sym_model_contr, sym_model_sys, X_init, X_target)
 	end
 	println("\nset_controller_reach! terminated with success")
 end
+
+# "Set" assumes sym_model_contr is empty initially... May fail if not respected
+function set_controller_safe!(sym_model_contr, sym_model_sys, X_init, X_safe)
+	@assert X_init.grid_space == X_safe.grid_space ==
```
Comment (blegat) [inline src/macros.jl:84]: Same comment, use `===`

**Comment by guberger** [comment]
Indeed. It is the "fixed-point algorithm" in the SCOTS paper

**Pull request merged**